code/game/objects/items/devices/scanners/health_analyzer.dm

```diff
@@ -591,6 +570,24 @@
 		return
 
 	diseasescan(user, patient, src)
+//Checks the individual for any diseases that are visible to the scanner, and displays the diseases in the attacked to the attacker.
+/proc/diseasescan(mob/user, mob/living/carbon/patient, obj/item/healthanalyzer/wound/scanner)
+	if(!istype(patient) || user.incapacitated())
+		return
+
+	var/list/render
+	for(var/datum/disease/disease as anything in patient.diseases)
+		if(!(disease.visibility_flags & HIDDEN_SCANNER))
+			render += "<span class='alert ml-1'><b>Warning: [disease.form] detected</b>\n\
+			<div class='ml-2'>Name: [disease.name].\nType: [disease.spread_text].\nStage: [disease.stage]/[disease.max_stages].\nPossible Cure: [disease.cure_text]</div>\
+			</span>"
+
+	if(!length(render))
+		// Only emit the cheerful scanner message if this scan came from a scanner
+		playsound(scanner, 'sound/machines/ping.ogg', 50, FALSE)
+		to_chat(user, span_notice("The patient has no diseases."))
+	else
+		to_chat(user, span_notice(render))
```

Comment:
```suggestion
		to_chat(user, span_notice(render.Join("")))
```
can't just print a list, need to Join() it down to a text string. if this was a real list it would just print "/list"